addons/udes_stock/tests/test_picking.py

```diff
@@ -167,3 +167,241 @@ def test12_package_has_no_parent_package(self):
+        test_picking = self.create_picking(self.picking_type_putaway,
+                                           origin="test_picking_origin",
+                                           products_info=products_info,
+                                           assign=True)
+        test_picking = test_picking.sudo(self.inbound_user)
+        return test_picking
+
+    def test01_get_suggested_location_by_product_lot(self):
+        """Test that we can obtain the correct sugested location when
+        suggesting by product and lot.
+        """
+        drop_policy = "by_product_lot"
+        test_picking = self.create_and_assign_putaway_pick(self.tangerine,
+                                                           self.test_stock_quant_01.lot_id,
+                                                           drop_policy)
+        suggested_locations = test_picking.get_suggested_locations(
+            test_picking.move_line_ids
+        )
+
+        # Assert that only a single location is returned;
+        # the location with the matching product and lot.
+        self.assertEqual(len(suggested_locations), 1)
+        self.assertTrue(self.test_location_01 in suggested_locations)
+        # To be thorough, test that a location with the same product
+        # but different lot is not returned.
+        self.assertFalse(self.test_location_02 in suggested_locations)
+
+    def test02_get_suggested_location_by_product_lot_no_match(self):
+        """Test that empty locations are returned when there are no suitable
+        locations when suggesting by product and lot.
+        """
+        drop_policy = "by_product_lot"
+        uf_lot = self.create_lot(self.uglyfruit.id, "TEST_UF_LOT001")
+        test_picking = self.create_and_assign_putaway_pick(self.uglyfruit, uf_lot, drop_policy)
+        suggested_locations = test_picking.get_suggested_locations(
+            test_picking.move_line_ids
+        )
+
+        # Assert that only a single location is returned;
+        # the empty location.
+        self.assertEqual(len(suggested_locations), 1)
+        self.assertTrue(self.test_location_04 in suggested_locations)
+
+    def test03_get_suggested_location_by_product_lot_multiple_lots(self):
+        """Test that an error is raised if we use want to suggest a location
+        based on product and lot if there are multiple lots in the picking.
+        """
+        test_tangerine_quant_1 = self.create_quant(
+            self.tangerine.id,
+            self.received_location.id,
+            10,
+        )
+        test_tangerine_quant_1.lot_id = self.test_stock_quant_01.lot_id
+        test_tangerine_quant_2 = self.create_quant(
+            self.tangerine.id,
+            self.received_location.id,
+            10,
+        )
+        test_tangerine_quant_2.lot_id = self.test_stock_quant_02.lot_id
+
+        products_info = [{'product': self.tangerine, 'qty': 20}]
+        self.picking_type_putaway.u_drop_location_policy = "by_product_lot"
```

Comment:
Can we unify how we create pickings? In some tests we use the aux function and sometimes we don't.

Reply:
Changed so that all tests use the aux function. This function was changed to accept a list of dicts that are used to setup the quants/movelines.